src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/PackageLibs.targets

```diff
@@ -179,7 +179,10 @@
       <!-- Include source files for symbol packages. -->
       <FilesToPackage Include="@(Compile->'%(FullPath)')">
         <TargetPath>src</TargetPath>
-        <TargetPath Condition="$([System.String]::new('%(FullPath)').StartsWith('$(ProjectRoot)'))">src\$([System.String]::new('%(FullPath)').Substring('$(ProjectRootLength)'))</TargetPath>
+        <!-- Be explicit about path seperators for the target path for source files. When building packages on *nix, the use of the DOS seperator
+             was causing us to include source files in our packages in a folder called src\src, instead of them being stripped. -->
+        <TargetPath Condition="$([System.String]::new('%(FullPath)').StartsWith('$(ProjectRoot)')) and '$(OSEnvironment)' == 'Windows_NT'">src\$([System.String]::new('%(FullPath)').Substring('$(ProjectRootLength)'))</TargetPath>
+        <TargetPath Condition="$([System.String]::new('%(FullPath)').StartsWith('$(ProjectRoot)')) and '$(OSEnvironment)' != 'Windows_NT'">src/$([System.String]::new('%(FullPath)').Substring('$(ProjectRootLength)'))</TargetPath>
```

Comment:
Could `src\` and `src/` be pulled out into a property so the condition and string logic doesn't have to be duplicated?